tools/tests/com.liferay.ide.project.core.tests/src/com/liferay/ide/project/core/tests/modules/jsf/NewLiferayJSFModuleProjectOpTests.java

```diff
@@ -0,0 +1,209 @@
+    {
+        NewLiferayJSFModuleProjectOp existedProjectop = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        existedProjectop.setProjectName( "Test3" );
+        existedProjectop.setProjectProvider( "maven-jsf-module" );
+        Status exStatus = NewLiferayJSFModuleProjectOpMethods.execute(
+            existedProjectop, ProgressMonitorBridge.create( new NullProgressMonitor() ) );
+        assertTrue( exStatus.ok() );
+
+        NewLiferayJSFModuleProjectOp newBuildTypeop = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        DefaultValueService buildTypeDefaultService =
+            newBuildTypeop.getProjectProvider().service( DefaultValueService.class );
+        assertEquals( buildTypeDefaultService.value(), "maven-jsf-module" );
+    }
+
+    @Test
+    public void testNewLiferayJSFModuleProjectOpJsfTemplate() throws Exception
+    {
+        String[] jsfTemplateArray = { "JSF Standard", "Liferay Faces Alloy", "ICEFaces", "PrimeFaces", "RichFaces" };
+        NewLiferayJSFModuleProjectOp op = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        PossibleValuesService jsfTemplateValuesService = op.getJsfTemplateName().service( PossibleValuesService.class );
+        Set<String> jsfTemplates = jsfTemplateValuesService.values();
+
+        ArrayList<String> jsfTemplatesList = new ArrayList<String>();
+        for( String jsfTemplate : jsfTemplates )
+        {
+            jsfTemplatesList.add( jsfTemplate );
+        }
+
+        String[] templateArray = jsfTemplatesList.toArray( new String[2] );
+
+        assertArrayEquals( templateArray, jsfTemplateArray );
+    }
+
+    @Test
+    public void testNewLiferayJSFModuleStandardAloneProject() throws Exception
+    {
+        NewLiferayJSFModuleProjectOp mavenProjectop = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        mavenProjectop.setProjectName( "Test4" );
+        mavenProjectop.setProjectProvider( "maven-jsf-module" );
+        Status mavenProjectStatus = NewLiferayJSFModuleProjectOpMethods.execute(
+            mavenProjectop, ProgressMonitorBridge.create( new NullProgressMonitor() ) );
+        assertTrue( mavenProjectStatus.ok() );
+
+        IProject mavenProject = CoreUtil.getProject( "Test4" );
+
+        assertNotNull( mavenProject );
+
+        NewLiferayJSFModuleProjectOp graldeProjectop = NewLiferayJSFModuleProjectOp.TYPE.instantiate();
+        graldeProjectop.setProjectName( "Test5" );
+        graldeProjectop.setProjectProvider( "gradle-jsf-module" );
+        Status gradleProjectStatus = NewLiferayJSFModuleProjectOpMethods.execute(
+            graldeProjectop, ProgressMonitorBridge.create( new NullProgressMonitor() ) );
+        assertTrue( gradleProjectStatus.ok() );
+
+        IProject gradleProject = CoreUtil.getProject( "Test5" );
+
+        assertNotNull( gradleProject );
+    }
+
+    @Test
+    public void testNewLiferayJSFModuleWorkspaceProject() throws Exception
```

Comment:
assert the project builds with no errors or markers like in other project tests.s